### ISSUE TITLE
Consider prioritizing `array_api` subsets

### ISSUE THREAD

**Issue by HaoZeke** [lcompilers/lpython#1662]
The [array_api specification](https://data-apis.org/array-api/latest/) is a good target since it is common to a bunch of tensor libraries. Eventually, we shouldn't implement `pytorch` and `numpy` operations the same way but as a first compatible pass it would make sense.

**Comment by certik** [comment]
Yes, that's a good starting subset of NumPy that we should support.

**Comment by certik** [comment]
I think this is almost a duplicate of https://github.com/lcompilers/lpython/issues/1458, but let's keep it open, as to what to prioritize from NumPy.
